---
layout: default
title: Research highlights
---

### Spatial transcriptomics and single cell analysis (2020-present)

Spatial transcriptomics enables the discovery of spatial organization of cell types, cell states, and biomarkers. I am involved in multiple research projects examining spatial patterns of gene expression based on 10x's Visium technology. 

Typical bioinformatics analyses performed include: 
- Unsupervised clustering and identification of cluster-specific genes
- Cell segementation and imaging analysis
- Identification of spaital variable genes
- Integration with scRNA-seq data
- Neighborhood enrichment analysis
- Cell type prediction/annotation
- Cell-cell communication analysis
- Pseudotime analysis

I am co-I with Drs. Arthur Beyder & Gianrico Farrugia on a federal grant investigating Mechanotransduction in Intestinal Smooth Muscle Cells using spatial transcriptomics technology (R01 DK 52766)

### Benign breast disease (2018-present)

Benign breast disease (BBD), which includes non-proliferative, proliferative and proliferative lesions with atypia, is viewed as a nonobligate precursor stage in the development of breast cancer, and is associated with an increased risk of invasive BC, particularly in those with proliferative or atypical lesions. I am part of Mayo Clinic's BBD analytical team. Below are some projects that I am involved in.

- Cell type deconvolution
Bulk samples of heterogeneous mixtures only represent averaged expression levels. I am involved in evaluating strategies for accurate cell-type deconvolution for BBD FFPE RNA-seq samples.

- Genomics risk factors associated with BBD
We performed high-coverage DNA sequencing and investigated gene-level mutation associations with breast cancer risk. 

- Sequencing and bioinformatics considerations for FFPE tissues
We evaluated DNA&RNA sequencing protocols tailored for FFPE tissues and recommended sequencing considerations and bioinformatics strategies for processing FFPE samples. 

### CNV/SV detection and prioritization (2019-present)

Many methods/callers have been developed to detect CNV/SVs, but there is a lack of strategy in combining results from multiple callers. We are working on developing a common data format in harmonizing and combining results from multiple callers, followed by filtering based on variant attributes (Supporting reads & Genomics mappability, etc.). 

### Xenografts RNA-seq data analysis (2018-present)

As patient derived xenograft (PDX) models are increasingly used for preclinical drug development, strategies to account for the nonhuman component of PDX RNA expression data are critical to its interpretation. I have developed and evaluated a bioinformatics pipeline to separate donor tumor and mouse stroma transcriptome profiles using PDX samples from ovarian cancer.

### Differential analysis (2013-2017)

- RNA-seq
I developed an algorithm (XBSeq & XBSeq2), a differential expression analysis algorithm for RNA-seq data that takes non-exonic mapped reads into consideration.

- MBDcap-seq
A similar statistical model (MBDDiff) was applied to MBDcap-seq data for testing differential methylation levels of promoter regions.

For a list of all my publications & abstracts, you can view my google scholar profile [here](https://scholar.google.com/citations?user=KB7SiiwAAAAJ&hl=en)