<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
  <head>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-66197207-1', 'auto');
  ga('send', 'pageview');

</script>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <meta name="author" content="Yuanhang Liu" />
    
    <title>Comparison of differential expression methods for RNA-seq</title>
    
    <link rel="shortcut icon" href="/favicon.ico">
    <link href="/atom.xml" rel="alternate" title="Takes time to be twisted, or not" type="application/atom+xml" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.2.0/css/font-awesome.min.css">
    <link rel="stylesheet" href="/media/css/style.css">
    <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/retina.js/1.3.0/retina.min.js"></script>
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/8.4/styles/github.min.css">
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/8.4/highlight.min.js"></script>
    <script type="text/javascript"> hljs.initHighlightingOnLoad(); </script>
  </head>
  <body>
      <div id="main" role="main">
        <header>
          <div id="header">
            <h1><a title="Takes time to be twisted, or not" class="" href="/">Takes time to be twisted, or not</a></h1>
          </div>
          <nav>
            
            <span><a title="Archive" href="/archive.html"><i class="fa fa-list-ul"></i></a></span>
            
            <span><a title="Tags" href="/tags.html"><i class="fa fa-tags"></i></a></span>
            
            <span><a title="About" href="/about.html"><i class="fa fa-user"></i></a></span>
            
            <span><a title="Gallery" href="https://www.flickr.com/photos/133563883@N05/"><i class="fa fa-film"></i></a></span>
            
          </nav>
        </header>
        <div id="content">
        <article>
  <section class="title">
    <h2>Comparison of differential expression methods for RNA-seq </h2>
  </section>
  <section class="meta">
  <span class="time">
    <time datetime="2015-08-02">2015-08-02</time>
  </span>
  
  <span class="tags">
    
    <a href="/tags.html#RNA-seq" title="RNA-seq">#RNA-seq</a>
    
    <a href="/tags.html#Genomics" title="Genomics">#Genomics</a>
    
  </span>
  
  </section>
  <section class="post">
  <p>This is mainly the summary results from ‘Comprehensive evaluation of differential gene expression analysis methods for RNA-seq data’ <a href="http://www.genomebiology.com/2013/14/9/r95#B27">http://www.genomebiology.com/2013/14/9/r95#B27</a>.</p>

<h2 id="normalization">Normalization</h2>

<ul>
  <li>reads per kilobase per million reads (RPKM)</li>
</ul>

<script type="math/tex; mode=display">RPKM = ({10^9} \cdot C)/(N \cdot L)</script>

<p>Where C stands for number of reads mapped to a gene (Number of fragments in terms of FPKM); N stands for total library size; L stands for gene length.</p>

<p>RPKM and FPKM has some obvious limitations. RPKM and FPKM are both propotional to the real abundance of a transcript/isoform, however, for different experiments, the propotional ratio is not constant. This means that you need to do further normalization after RPKM or FPKM.</p>

<p>All these methods have one limitations in common: It is relative abundance, which means the expression changes in one gene can also affect the relative abundance of other genes.</p>

<ul>
  <li>TPM</li>
</ul>

<script type="math/tex; mode=display">TPM = p\cdot{10^6}</script>

<p>Or can also be formulated by:</p>

<script type="math/tex; mode=display">TPM = \frac{FPKM\cdot{10^6}}{\sum FPKM}</script>

<p>where p stands for the probablity of selecting a fragment from one transcirpt among all read counts. Different from RPKM/FPKM, TPM maesures estimated abundance rather than relative abundance. TPM is generally preferred than RPKM/FPKM</p>

<ul>
  <li>DESeq’s sizeFactors</li>
</ul>

<script type="math/tex; mode=display">\hat{s_j} = \underset{i}{median}\frac{k_{ij}}{\left(\prod_{v=1}^m k_{iv}\right)^{\frac 1m}}</script>

<ul>
  <li>Cuffdiff</li>
</ul>

<p>Similar to DESeq, firstly intro-condition scaling factor, secondly, across condition scaling factor.</p>

<ul>
  <li>Trimmed mean of M values (TMM)</li>
</ul>

<p>TMM is implemented in edgeR which computes a scaling factor between two experiments by using the weighted average of the subset of genes after excluding genes that exhibit high average read counts and genes that have large differences in expression.</p>

<p>baySeq implements similar method as TMM; It uses genes with upper quantile in terms of expression levels to calculate scaling factor</p>

<p>PoissonSeq use a subset of genes (Least differentially expressed between conditions) to calculate a scaling factor for normalization.</p>

<ul>
  <li>Quantile</li>
</ul>

<p>Very widely used normalization method for microarray, implemented in limmma; Basically ensure that the empirical distribution across all samples is the same.</p>

<h2 id="statistical-modeling">Statistical modeling</h2>

<ul>
  <li>Poisson</li>
</ul>

<p>PoissonSeq</p>

<ul>
  <li>Negative binomial</li>
</ul>

<p>To account for over-dispersion problem. DESeq and edgeR, DESeq and edgeR differs in terms of estimation of dispersion parameter (squared coefficient of variation).</p>

<p>DESeq firstly models mean-variance relationship to get a better estimation of dispersion (squared coefficient of variation). For smaller number of replicates, additional apply a bias correction procedure for dispersion estimates.</p>

<p>edgeR firstly estimate a common dispersion of all the genes, then applied a weighted maximum liklihood estimation of the weighted paramter of common dispersion and gene-wise dispersion, so that if the dispersion are very variable (different) across all the genes, the weight parameter should be very small, otherwise, it should be very big (to a common dispersion estimates)</p>

<script type="math/tex; mode=display">WL(\phi_g) = l_g(\phi_g) + \alpha \cdot l_C(\phi_g)</script>

<p>Cufdiff models single isoform and multi-isoform differently; For single isoform, cufdiff applies a similar method as DESeq. Multi-isoform variance is modeled by a mixture model of negative binomials using the beta distribution parameters as mixture weights.</p>

<p>baySeq implements a full Bayesian model of negative binomial distributions in which the prior probability parameters are estimated by numerical sampling from the data</p>

<h2 id="test-for-differential-expression">Test for differential expression</h2>

<ul>
  <li>Fisher exact test for NB distribution</li>
</ul>

<p>Both DESeq and edgeR adopted a NB flavor fisher exact test</p>

<script type="math/tex; mode=display">p = \frac{\sum_{a+b=K_S \; p(a,b) \le p(K_A, K_B)} p(a,b)}{\sum_{a+b=K_S} p(a,b)}</script>

<ul>
  <li>t test or moderated t test</li>
</ul>

<p>Cuffdiff computes test statistics:</p>

<script type="math/tex; mode=display">T = \frac{E[log(y)]}{Var[log(y)]}</script>

<p>This ratio approximately follows a normal distribution, a t test is applied to identify DEGs</p>

<p>limma uses a moderated t-statistic to compute P values in which both the standard error and the degrees of freedom are modified</p>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-66197207-1', 'auto');
  ga('send', 'pageview');

</script>

<h2 id="references">references</h2>

<p>Lior Pachter’s talk <a href="http://www.homolog.us/blogs/blog/2013/11/22/cshl-keynote-talk-lior-pachter/">http://www.homolog.us/blogs/blog/2013/11/22/cshl-keynote-talk-lior-pachter/</a></p>

<p>Discussion of RPKM, FPKM and TPM on google groups <a href="https://groups.google.com/forum/#!topic/rsem-users/GRyJfEOK1BQ">https://groups.google.com/forum/#!topic/rsem-users/GRyJfEOK1BQ</a></p>


  </section>
  
</article>

<div id="disqus_thread"></div>
<script type="text/javascript">
    /* * * CONFIGURATION VARIABLES * * */
    // Required: on line below, replace text in quotes with your forum shortname
    var disqus_shortname = 'personalblogliuy';
    
    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript" rel="nofollow">comments powered by Disqus.</a></noscript>


        </div>
        <footer>
          <div>
            
            &copy; 2015 ~ 2015 Yuanhang Liu | powered by jekyll | themed by <a href="http://lhzhang.com" title="sext vi">sext vi</a> | fork <a href="https://github.com/waynezhang/blog" title="fork me">me</a>
          </div>
        </footer>
      </div> <!-- main -->
  </body>
</html>
